### PR TITLE
1367 programs wrong label / name for the programs

### LINKED ISSUE #1367: Programs: wrong label / name for the programs
## What went wrong? 😲
 

encounter list: 
Screen Shot 2023-03-16 at 1 10 36 PM

encounter detailed view's headers: 
Screen Shot 2023-03-16 at 1 10 05 PM

## Expected behaviour 🤔

correct label for programs should be:

`HIV Care & Treatment` instead of `HIVCareProgram`
`HIV Testing & Counselling` instead of `HIVTestingProgram`

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1367 

# 👩🏻‍💻 What does this PR do? 
Use document registry to get program name

### CODE REVIEW

File: client/packages/system/src/Encounter/DetailView/Toolbar.tsx [hunk 1]
```diff
@@ -95,10 +103,16 @@ export const Toolbar: FC<ToolbarProps> = ({ encounter, onChange }) => {
                 }
               />
             </Box>
-            <Box display="flex" gap={1}>
+            <Box display="flex" gap={0.5}>
```
Comment (roxy-dao) [inline client/packages/system/src/Encounter/DetailView/Toolbar.tsx:106]: Had to make the program text field wider, so have adjusted the gaps to line up the components

File: client/packages/system/src/Encounter/ListView/columns.ts [hunk 2]
```diff
@@ -9,13 +9,20 @@ import { useFormatDateTime } from '@common/intl';
...
   // just take the name of the first event
   return events[0]?.data ?? '';
 };
 
+const getProgramName = (program: string) => {
+  const { data } = useDocumentRegistry.get.documentRegistryByType(program);
+
+  return data?.[0]?.name;
+};
+
```
Comment (clemens-msupply) [inline client/packages/system/src/Encounter/ListView/columns.ts:25]: This probably works and is legal but I find it hard reason about it... What I mean is that the hook is hidden in this function but we have to make sure to always call all hooks to not confuse React. Possible problem is that you may return early from `useEncounterListColumns` before calling `getProgramName`. That this is not allowed is not obvious, e.g. for a different dev...

I would just inline this method, i.e. call `useDocumentRegistry.get.documentRegistryByType` straight after `useFormatDateTime`....

Comment (roxy-dao) [reply client/packages/system/src/Encounter/ListView/columns.ts]: [90e5fa1](https://github.com/openmsupply/open-msupply/pull/1417/commits/90e5fa118615508d4be9b843da99d509e6e69816)

File: client/packages/system/src/Encounter/ListView/columns.ts [hunk 3]
```diff
@@ -40,6 +41,13 @@ export const useEncounterListColumns = ({
     {
       key: 'program',
       label: 'label.program',
+      accessor: ({ rowData }) => {
+        const { data } = useDocumentRegistry.get.documentRegistryByType(
+          rowData.program
+        );
```
Comment (clemens-msupply) [inline client/packages/system/src/Encounter/ListView/columns.ts:47]: Sorry was too quick to review previous commit and missed how `accessor()` works...

Still have the same concern as previously.

@mark-prins is it legal to call a hook in `accessor`? Most uses seem fine but find it still a bit hacky? We could allow it and then rename `accessor` to `useAccessor` and refactor some of the usages to move them up a little e.g. in `packages/common/src/ui/layout/tables/components/index.tsx`....

Alternative would be to fetch all registry entries (which still should be cached anyway from previous calls) and then pick the correct one here...

Comment (clemens-msupply) [reply client/packages/system/src/Encounter/ListView/columns.ts:47]: Had a quick check with Mark on Fri, loading the full registry and then selecting the correct entry in the `accessor` should be the best way to do here...

Another though on it, having hooks as parameters, e.g. in the accessor, is IMHO not a good idea, e.g. you could write code that changes the hook on the fly which could lead to the same problem that hooks are not always called in the right order.

Comment (roxy-dao) [reply client/packages/system/src/Encounter/ListView/columns.ts:47]: Load the full registry in the column, then use the accessor to get the correct entry? Is that correct?

File: client/packages/common/src/hooks/useQueryParams/useQueryParamsStore.ts [hunk 4]
```diff
@@ -125,6 +125,18 @@ export const createQueryParamsStore = <T extends RecordWithId>({
...
+        key: string,
+        condition: FilterByConditionByType['stringArray'],
+        value: string[]
+      ) => {
+        if (value.length === 0) {
+          get().filter.onClearFilterRule(key);
+        } else {
+          setFilterBy(set)({ [key]: { [condition]: value } });
+        }
+      },
```
Comment (clemens-msupply) [inline client/packages/common/src/hooks/useQueryParams/useQueryParamsStore.ts:138]: Are these changes needed?

**Review by clemens-msupply — APPROVED** [review]
thanks you looks good!

**Pull request merged**